ip-location-service.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+
+const path = require("path");
+const reader = require("@maxmind/geoip2-node").Reader;
+const AppConstants = require("./app-constants");
+
+let locationDb, timestamp;
+
+async function openLocationDb() {
+  if (locationDb && isFresh()) {
+    console.warn("Location database opened less than 24hrs: returning original instance.");
+    return locationDb;
+  }
+
+  try {
+    const dbPath = path.join(AppConstants.GEOIP_GEOLITE2_PATH, AppConstants.GEOIP_GEOLITE2_CITY_FILENAME);
+    locationDb = await reader.open(dbPath);
+  } catch (e) {
+    return console.warn("Could not open location database:", e.message);
```

Comment:
I'm enjoying insight into how others might read my code 😊 .  

Yes, returning the object in `openLocationDb()` is a "benign" remnant of an earlier attempt.  I see how it can be confusing – all returns could just have easily been `undefined`.  I can/might just remove the returned values.

My intention was to use locationDb as a private singleton.  I don't believe it's global.